osu.Game/Screens/SelectV2/BeatmapSetPanel.cs

```diff
@@ -69,11 +289,159 @@ protected override void PrepareForUse()
             Debug.Assert(Item != null);
             Debug.Assert(Item.IsGroupSelectionTarget);
 
-            var beatmapSetInfo = (BeatmapSetInfo)Item.Model;
+            var beatmapSet = (BeatmapSetInfo)Item.Model;
+
+            // Choice of background image matches BSS implementation (always uses the lowest `beatmap_id` from the set).
+            background.Beatmap = beatmaps.GetWorkingBeatmap(beatmapSet.Beatmaps.MinBy(b => b.OnlineID));
+
+            titleText.Text = new RomanisableString(beatmapSet.Metadata.TitleUnicode, beatmapSet.Metadata.Title);
+            artistText.Text = new RomanisableString(beatmapSet.Metadata.ArtistUnicode, beatmapSet.Metadata.Artist);
+            updateButton.BeatmapSet = beatmapSet;
```

Comment:
Likewise, this is never cleaned up in `FreeAfterUse` which means the internal tracking is not cleaned up. Probably want to clean these up.

Reply:
I will change this to be cleaned up on `FreeAfterUse` alongside any other references, but similar to https://github.com/ppy/osu/pull/31774#discussion_r1940953285, there is nothing in the button tracking the internal state of a beatmap as far as I can see?